File: p5/p5.pl
#!/usr/bin/perl
use strict;

sub gcd{
    my ($a, $b, @r) = @_;
    @r ? gcd($a, gcd($b,@r)) : $b == 0 ? $a : gcd($b, $a % $b)
}

sub lcm{
    my ($a, $b, @r) = @_;    
    @r ? lcm($a, lcm($b, @r)): $a * $b / gcd($a, $b)
}

print "Answer".lcm(1 .. 20)."\n";

File: p5/p5.py
#!/usr/bin/python

# Euclid's algorithm
def gcd(a, b): return b and gcd(b, a % b) or a 
def lcm(a, b): return a * b / gcd(a, b)

answer = 1
for i in xrange(1,21): answer = lcm(answer,i)

print "Answer %d " % answer
